app/models/waiting_list.rb

```diff
@@ -3,30 +3,33 @@
 class WaitingList < ActiveRecord::Base
   belongs_to :holder, polymorphic: true
 
-  def remove(entry_id)
-    update_column :entries, entries - [entry_id]
+  def remove(item)
+    update_column :entries, entries - [item.id]
   end
 
-  def add(entry_id)
+  def add(item)
+    item.can_be_waitlisted! # can_be_waitlisted! raises an error if it fails
```

Comment:
Ruby trick: You can do `item.try(:can_be_waitlisted!)` which (obviously) _tries_ to call it if the method is defined, but doesn't explode if you forget to implement it.

Reply:
Implemented